x-pack/platform/plugins/shared/inference/server/chat_complete/adapters/inference/inference_adapter.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type OpenAI from 'openai';
+import { from, identity, switchMap, throwError } from 'rxjs';
+import { isReadable, Readable } from 'stream';
+import { createInferenceInternalError } from '@kbn/inference-common';
+import { eventSourceStreamIntoObservable } from '../../../util/event_source_stream_into_observable';
+import type { InferenceConnectorAdapter } from '../../types';
+import {
+  parseInlineFunctionCalls,
+  wrapWithSimulatedFunctionCalling,
+} from '../../simulated_function_calling';
+import {
+  toolsToOpenAI,
+  toolChoiceToOpenAI,
+  messagesToOpenAI,
+  processOpenAIStream,
+} from '../openai';
+
+export const inferenceAdapter: InferenceConnectorAdapter = {
+  chatComplete: ({
+    executor,
+    system,
+    messages,
+    toolChoice,
```

Comment:
The adapter working with the inference connector. Very similar to the existing openAI adapter, which is why most of the in/out processing logic has been factorized.